### PR TITLE
Mock: Added a registry of provider dependent files

### PR DESCRIPTION
**Pull request by andy-maier**
This is in support of https://github.com/pywbem/pywbemtools/issues/748.
For details, see the commit message.

### CODE REVIEW

**Comment by andy-maier** [comment]
We agreed to pursue this path.

File: pywbem_mock/_providerdependentregistry.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+from pywbem._utils import _format
+
+__all__ = ['ProviderDependentRegistry']
+
+
+class ProviderDependentRegistry(object):
+    """
+    A registry for provider dependent files in context of a mock script.
+
+    This registry allows registering additional dependent files in context of
```
Comment (KSchopmeyer) [inline pywbem_mock/_providerdependentregistry.py:31]: Should we be more complete here to state the reason for register and lookup.  Thus to register dependent files when a mock script is cached and to look up and validate the mock_script file dependencies when the cached mock script is loaded.

File: docs/mockwbemserver.rst [hunk 2]
```diff
@@ -1179,6 +1179,39 @@ User-defined method providers
...
+-------------------------------------
+
+A faked WBEM connection provides a registry for provider dependent files
+in its :attr:`~pywbem_mock.FakedWBEMConnection.provider_dependent_registry`
+property of type :class:`pywbem_mock.ProviderDependentRegistry`.
+
+This registry can be used by callers to register and look up the path names of
+additional dependent files of a mock script, in context of that mock script.
+
+The pywbemtools project makes use of this registry for validating whether its
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:1194]: Change "makes use of this registry" to "uses this registry"

File: docs/mockwbemserver.rst [hunk 3]
```diff
@@ -1179,6 +1179,39 @@ User-defined method providers
...
+.. _`Registry for provider dependent files`:
+
+Registry for provider dependent files
+-------------------------------------
+
+A faked WBEM connection provides a registry for provider dependent files
+in its :attr:`~pywbem_mock.FakedWBEMConnection.provider_dependent_registry`
+property of type :class:`pywbem_mock.ProviderDependentRegistry`.
+
+This registry can be used by callers to register and look up the path names of
```
Comment (KSchopmeyer) [inline docs/mockwbemserver.rst:1191]: Should we explain why register and lookup are used?

**Review by KSchopmeyer — COMMENTED** [review]
Some editorial comments. Otherwise approved

File: pywbem_mock/_providerdependentregistry.py (hunk 1 above)
Comment (andy-maier) [reply pywbem_mock/_providerdependentregistry.py:31]: The possible use for validating a cache in pywbemtools is already described in the mockwbemserver.rst file just before this class is shown, as:
```
The pywbemtools project makes use of this registry for validating whether its
mock cache is up to date w.r.t. additional dependent files a mock script has
used.
```

File: docs/mockwbemserver.rst (hunk 3 above)
Comment (andy-maier) [reply docs/mockwbemserver.rst:1191]: The next sentence does that:
```
The pywbemtools project makes use of this registry for validating whether its
mock cache is up to date w.r.t. additional dependent files a mock script has
used.
```

**Comment by andy-maier** [comment]
I think both comments had been already addressed.

**Pull request merged**